### DISCUSSION TITLE
Unwanted MIDI notes being sent constantly

### DISCUSSION THREAD

**Discussion by Alex0327** [tttapa/Control-Surface#1022] [Q&A]
Using the NoteButtonMatrix.ino example file on Arduino Uno and Mega, I am getting many random notes playing for a fraction of a second constantly. I had modified the code for the numbers of rows and columns and the notes i needed and get the same result. It only plays the notes that are written in the code. Here are screenshots of the midi data being sent. Can someone help me fix this issue? Keep in mind I am a beginner when it comes to coding and Arduino.
![midi1](https://github.com/tttapa/Control-Surface/assets/11903755/42863617-271a-4987-9761-d586d2c42a2a)
![midi2](https://github.com/tttapa/Control-Surface/assets/11903755/d8263284-a2b0-447a-a4f7-5e33c831bd1e)

Update 1: Here is the slightly modifed example code to meet my needs. I have tried different pins on the arduino mega with no change. My project is to MIDIfy a 32 note organ pedalboard. Pictured is the orginal reed switches with diodes in a 6 x 7 matrix array, fully working. With this current setup, I can play each note by holding a magnet to each sensor and the correct note plays when the magnet is near and turns off as the magnet is removed. I am not using MIDI over the usb cable. I have never gotten that to work with other Control Surface code. Instead I built my own MIDI I/O interface for the arduino based on the schematics from notesandvolts.com, pictured below. I have gotten my MIDI interface to work with other Control Surface code with no extraneous notes. The switches for the pedalboard function perfectly with this code. The only problem is the constant, very short, multiple midi notes per second. I have to have a MIDI port for the pedalboard to plug into my organ.

`#include <Control_Surface.h>

USBMIDI_Interface midi;

// The note numbers corresponding to the buttons in the matrix
const AddressMatrix<7, 6> addresses {{
  {0, 0, 0, 0, 0, 24},
  {39, 40, 37, 38, 41, 42},
  {45, 46, 43, 44, 47, 48},
  {51, 52, 49, 50, 53, 54},
  {0, 0, 55, 0, 0, 0},
  {27, 28, 25, 26, 29, 30},
  {33, 34, 31, 32, 35, 36},
}};

NoteButtonMatrix<7, 6> buttonmatrix {
  {29, 37, 41, 43, 45, 47, 49}, // row pins
  {53, 51, 39, 35, 33, 31},    // column pins
  addresses,    // address matrix
  Channel_1,    // channel and cable number
};

void setup() {
  Control_Surface.begin();
}

void loop() {
  Control_Surface.loop();
}`

![20240317_132510](https://github.com/tttapa/Control-Surface/assets/11903755/667c4065-ace8-40f1-b714-28b1c4c80f24)
![20240317_132514](https://github.com/tttapa/Control-Surface/assets/11903755/f6bcb6cd-b9e9-4c7a-9180-71e4335016fd)
![20240317_132527](https://github.com/tttapa/Control-Surface/assets/11903755/60c475d0-fa78-4d08-be84-f86753500bbd)
![MIDI_Input_Schematic](https://github.com/tttapa/Control-Surface/assets/11903755/b8d0d8e8-8be8-4f9e-b6fb-326102e2a3a6)
![SCHEMATIC_Midi_Output](https://github.com/tttapa/Control-Surface/assets/11903755/ae082dd4-df55-4d9b-9bd2-0df6e53bcef5)

**Comment by tttapa** [comment]
Please post your code and a picture of the wiring.

**Reply by Alex0327** [reply]
Any help would be appreciated.

**Comment by tttapa** [comment]
The wires you have are quite long, so their capacitance is much higher than usual, and they might be picking up a lot of noise. You should definitely use stronger pull-up resistors than the built-in ones. You could also try increasing the line select delay:

https://github.com/tttapa/Control-Surface/blob/d16582db32cad30a40e638556c096a92bec45fca/src/AH/Settings/Settings.hpp#L78-L81

Then also enable this delay for AVR here:

https://github.com/tttapa/Control-Surface/blob/d16582db32cad30a40e638556c096a92bec45fca/src/AH/Hardware/ButtonMatrix.ipp#L25-L27

(i.e. remove the `!defined(__AVR__) &&`).

**Reply by Alex0327** [reply]
I didn't bother with the resistors, just changed the delay to 20 microseconds and everything works perfectly with no extra notes being sent. Thank you so much!

**Reply by Alex0327** [reply]
It worked once. Then, I unplugged the pins and set it up again and I was back to the original problem. I tried longer delay values and it lessened how many midi notes were being sent, but didn't fix the issue. I'm a beginner, where exactly would i need to put the resistors?
